catalyst_voices_packages/catalyst_key_derivation/rust/src/api/key_derivation/mod.rs

```diff
@@ -0,0 +1,328 @@
+//! Cardano deterministic key hierarchy using BIP-0039 module.
+//!
+//! This module provides functions necessary to handle deterministic key derivation
+//! using BIP-0039 mnemonics.
+
+use bip32::DerivationPath;
+use bip39::Mnemonic;
+pub use ed25519_bip32::{DerivationIndex, DerivationScheme, Signature, XPrv, XPub};
+use flutter_rust_bridge::spawn_blocking_with;
+use hmac::Hmac;
+use pbkdf2::pbkdf2;
+use sha2::Sha512;
+
+use crate::frb_generated::FLUTTER_RUST_BRIDGE_HANDLER;
+
+/// Extended private key bytes type.
+/// Compose of:
+/// - 64 Bytes: extended Ed25519 secret key
+/// - 32 Bytes: chain code
+pub type XPrvBytes = [u8; 96];
+
+/// Extended public key bytes type.
+pub type XPubBytes = [u8; 64];
+
+/// Signature bytes type.
+pub type SignatureBytes = [u8; 64];
```

Comment:
What do you think about using `newtypes` instead of `type aliases`? It could help prevent accidentally mixing up the different key types.